akka-http-core/src/main/scala/akka/http/scaladsl/ConnectionContext.scala

```diff
@@ -24,31 +29,98 @@ trait ConnectionContext extends akka.http.javadsl.ConnectionContext {
 object ConnectionContext {
   // ConnectionContext
   //#https-context-creation
+  def https(createSSLEngine: Option[(String, Int)] => () => SSLEngine): HttpsConnectionContext =
+    new HttpsConnectionContext(Right(o => Engine(createSSLEngine(o))))
+
+  def httpsServer(sslContext: SSLContext): HttpsConnectionContext =
+    new HttpsConnectionContext(Right({
+      case None => Engine(() => {
+        val engine = sslContext.createSSLEngine()
+        engine.setUseClientMode(false)
+        engine
+      })
+      case Some((host, port)) => Engine(() => {
+        val engine = sslContext.createSSLEngine(host, port)
+        engine.setUseClientMode(false)
+        engine
+      })
+    }))
+
+  def httpsClient(context: SSLContext)(implicit system: ClassicActorSystemProvider): HttpsConnectionContext = {
+    val verifier = new DefaultHostnameVerifier(new AkkaLoggerFactory(system.classicSystem))
```

Comment:
Is there a specific reason we need this one?

Reply:
I'm not sure what you mean - without this hostname verification would not work (demonstrated by TlsEndpointVerificationSpec). Is there another implementation that might be more suitable?

Reply:
Hostname verification should work out of the box when `setEndpointVerificationMethod("https")` is used.

Reply:
Indeed :+1: